databuilder/dsl.py

```diff
@@ -127,6 +128,11 @@ def __init__(self, qm_table: BaseTable):
         """
         self.qm_table = qm_table
 
+    @classmethod
+    def from_contract(cls, contract):
+        qm_table = Table(contract._name)
+        return cls(qm_table)
```

Comment:
The "root" (table) frames don't have any different behaviour from other frames, but they clearly play a special role and that's not clear in the current object model. If we had `EventTable` and `PatientTable` as subclasses then it would clarify that role and they could have constructors that take contracts rather than a special class method.

Reply:
Addressed in 530a934a9af294cc5b0de618d5368a72a94791ff.